src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.Inventory;
+
+public class PurchaseOperation implements FruitOperationHandler {
+    @Override
+    public void executeOperation(FruitTransaction transaction, Inventory inventory) {
+        int currentBalance = inventory.getInventory().getOrDefault(transaction.getFruit(), 0);
+        int newBalance = currentBalance - transaction.getQuantity();
+
+        if (newBalance < 0) {
```

Comment:
This check is correct according to the checklist, which states that the result balance should be positive. Good job on throwing a `RuntimeException` with an informative message if the balance is negative.